src/libcore/char.rs

```diff
@@ -420,7 +430,8 @@ pub fn encode_utf16_raw(mut ch: u32, dst: &mut [u16]) -> Option<usize> {
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct EscapeUnicode {
     c: char,
-    state: EscapeUnicodeState
+    state: EscapeUnicodeState,
+    offset: usize,
```

Comment:
`offset` is the name of the variable in the original code; I agree it does not provide much information, maybe `digit_index` might convey better its meaning